mllib/src/main/scala/org/apache/spark/ml/source/image/ImageFileFormat.scala

```diff
@@ -71,8 +71,8 @@ private[image] class ImageFileFormat extends FileFormat with DataSourceRegister
       if (!imageSourceOptions.dropInvalid && requiredSchema.isEmpty) {
         Iterator(emptyUnsafeRow)
       } else {
-        val origin = file.filePath
-        val path = new Path(origin)
+        val origin = file.uriEncodedPath
+        val path = file.hadoopPath // TODO (URI)
```

Comment:
Can we file a JIRA and add a link here so that we don't forget ?

e.g.
```scala
// TODO(SPARK-XXXXX): <add short summary for the TODO here would be great>
val path = file.hadoopPath
```